### PR TITLE
aws-janitor-boskos: add clean time and process time metrics 

### PR DESCRIPTION
**Pull request by cpanato**
...to collect the response time when cleaning and for the entire process when doing the janitor process

Make the histogram and created the exponential buckets, I've spoken in the past with `Frederic Branczyk - @brancz` and he suggests following this approach since it is very similar to what I've implemented in the past then used the same 😄 

this is part of https://github.com/kubernetes-sigs/boskos/issues/13

opening this PR for feedback and see if sounds good, and then will do it for the other janitors

/assign @ixdy

### CODE REVIEW

File: cmd/aws-janitor-boskos/main.go [hunk 1]
```diff
@@ -132,8 +164,12 @@ func cleanResource(res *common.Resource) error {
 		}
 	}
 
-	duration := time.Since(start)
-
-	logrus.WithFields(logrus.Fields{"name": res.Name, "duration": duration.Seconds()}).Info("Finished cleaning")
+	collectMetric(start, res.Name, "clean")
+	logrus.WithFields(logrus.Fields{"name": res.Name, "duration": time.Since(start).Seconds()}).Info("Finished cleaning")
```
Comment (ixdy) [inline cmd/aws-janitor-boskos/main.go:168]: do you think we should include sweepCount somehow as well? might be interesting to understand if a longer duration is due to multiple sweeps.

File: cmd/aws-janitor-boskos/main.go [hunk 2]
```diff
@@ -44,18 +51,33 @@ var (
...
+
+	instrumentationOptions prowflagutil.InstrumentationOptions
 )
 
 const (
 	sleepTime = time.Minute
 )
 
+var (
+	cleaningTimeHistogram = prometheus.NewHistogramVec(prometheus.HistogramOpts{
```
Comment (ixdy) [inline cmd/aws-janitor-boskos/main.go:63]: nit: this could probably be included in the `var` block above.

**Review by ixdy — COMMENTED** [review]
Thanks! This generally LGTM, just one question.

File: cmd/aws-janitor-boskos/main.go (hunk 1 above)
Comment (cpanato) [reply cmd/aws-janitor-boskos/main.go:168]: That sounds a great idea, I will update this PR over the weekend
thanks for your review and feedback

File: cmd/aws-janitor-boskos/main.go (hunk 2 above)
Comment (cpanato) [reply cmd/aws-janitor-boskos/main.go:63]: missed the var block above 🤦

File: cmd/aws-janitor-boskos/main.go (hunk 1 above)
Comment (cpanato) [reply cmd/aws-janitor-boskos/main.go]: added, but not 100% if it is correct, when you have time can you please take a look and let me know? thanks!

File: cmd/aws-janitor-boskos/main.go [hunk 3]
```diff
@@ -116,24 +153,32 @@ func cleanResource(res *common.Resource) error {
...
 			time.Sleep(sweepSleepDuration)
 		}
 	}
 
-	duration := time.Since(start)
-
-	logrus.WithFields(logrus.Fields{"name": res.Name, "duration": duration.Seconds()}).Info("Finished cleaning")
+	sweepsGauge.WithLabelValues(res.Name).Set(float64(countSweeps))
+	collectMetric(start, res.Name, "clean")
+	logrus.WithFields(logrus.Fields{"name": res.Name, "duration": time.Since(start).Seconds(), "sweeps": countSweeps}).Info("Finished cleaning")
```
Comment (ixdy) [inline cmd/aws-janitor-boskos/main.go:177]: Looking at this code again, I now realize I was misunderstanding it before - I thought that the specified `sweepCount` was the *maximum* number of attempted sweeps (i.e. retries if there was an error), but actually, I see that it always sweeps that many times, regardless of error or success.

We could keep the metric here, though it's not nearly as meaningful as I thought it was (since it'll basically remain the same for a given configuration). if we keep it, we can just pass `sweepCount` instead of using the separate `countSweeps` variable though.

Comment (cpanato) [reply cmd/aws-janitor-boskos/main.go:177]: I can drop that for now and in a follow-up do a small refactor in this part to only retry when it failed, and then we add back the metric. WDTY?

Comment (ixdy) [reply cmd/aws-janitor-boskos/main.go:177]: I think part of the reason it does multiple passes is that the AWS janitor typically does **not** return errors if it fails to delete resources, under the assumption that a later pass might successfully delete the resource (e.g. if there is some lingering dependency that gets cleaned up at a later stage).

This probably isn't ideal, but we'd need to investigate further and fix things to achieve the desired "retry only on error" behavior.

TL;DR we probably want to leave the existing behavior of always sweeping multiple times. The metrics you're adding can help us determine if this is taking too long and we should look into optimizing this process. :)

Comment (cpanato) [reply cmd/aws-janitor-boskos/main.go:177]: @ixdy sounds good :)

update the code to reflect your feedback, thanks again

**Review by ixdy — APPROVED** [review]
